test/unified_format.py

```diff
@@ -68,6 +70,27 @@
 
 JSON_OPTS = json_util.JSONOptions(tz_aware=False)
 
+IS_INTERRUPTED = False
+WIN32 = sys.platform in ("win32", "cygwin")
+
+
+def interrupt_handler(signum, frame):
+    global IS_INTERRUPTED
+    # Set the IS_INTERRUPTED flag here and perform the necessary cleanup
+    # before actually exiting in workload_runner. This is because signals
+    # are handled asynchronously which can cause the interrupt handlers to
+    # fire more than once. Consequently, the handler itself should be
+    # re-entrant (invokable multiple times without needing to wait for prior
+    # invocations to return/complete) which is made possible by this pattern.
+    IS_INTERRUPTED = True
+
+
+if WIN32:
+    # CTRL_BREAK_EVENT is mapped to SIGBREAK
+    signal.signal(signal.SIGBREAK, interrupt_handler)
+else:
+    signal.signal(signal.SIGINT, interrupt_handler)
```

Comment:
Hmm I was expecting this code to live in the drivers-atlas-testing workload runner. What do you think about replacing this with a function that the workload can use:
```python
def interrupt_loop():
    global IS_INTERRUPTED
    IS_INTERRUPTED = True
```

Then we can use this api from the workload executor.